packages/grid/_modules_/grid/utils/tree/buildRowTree.ts

```diff
@@ -108,19 +113,24 @@ export const buildRowTree = (params: BuildRowTreeParams): GridRowGroupingResult
       }
 
       let keyConfig = fieldSubTree[key.toString()];
-      if (!keyConfig) {
-        nodeId =
-          depth === row.path.length - 1
-            ? row.id
-            : `auto-generated-row-${row.path
-                .map((groupingCriteria) => `${groupingCriteria.field}/${groupingCriteria.key}`)
-                .slice(0, depth + 1)
-                .join('-')}`;
+
+      if (keyConfig) {
+        if (depth === row.path.length - 1) {
+          params.onDuplicatePath?.(keyConfig.id, row.id, row.path);
```

Comment:
For the row grouping, this scenario should never occur unless the user has duplicated ids, which would throw another error before.